components/components-processing/processing-runtime/src/main/java/org/talend/components/processing/runtime/normalize/NormalizeUtils.java

```
@@ -0,0 +1,276 @@
+                                        outputChildSchema);
+                                outputRecord.set(field.name(), childRecord);
+                            }
+                        } else {
+                            outputRecord.set(field.name(), inputValue);
+                        }
+                    }
+                } else {
+                    // The current column can be a Record (an hierarchical sub-object) or directly a value.
+                    // If we are on a record, we need to recursively do the process
+                    // if we are on a object, we save it to the output.
+                    if (inputValue instanceof GenericData.Record) {
+                        // The sub-schema at this level is a union of "empty" and a record,
+                        // so we need to get the true sub-schema
+                        Schema inputChildSchema = AvroUtils.unwrapIfNullable(inputSchema.getField(field.name()).schema());
+                        Schema outputChildSchema = AvroUtils.unwrapIfNullable(outputSchema.getField(field.name()).schema());
+                        if (inputChildSchema.getType().equals(Schema.Type.RECORD)
+                                && outputChildSchema.getType().equals(Schema.Type.RECORD)) {
+                            Object childRecord = dupplicateRecord((IndexedRecord) inputValue, inputChildSchema,
+                                    outputChildSchema);
+                            outputRecord.set(field.name(), childRecord);
+                        }
+                    } else {
+                        outputRecord.set(field.name(), inputValue);
+                    }
+                }
+            } else {
+                // element not found => set to the value and its hierarchy to null
+                outputRecord.set(field.name(), generateEmptyRecord(outputSchema, field.name()));
```

This should not append is wrong.
Could you throw an error?